Packs/CyberArkIdentity/Integrations/CyberArkIdentityEventCollector/CyberArkIdentityEventCollector.py

```diff
@@ -0,0 +1,231 @@
+        )
+
+        response = self.call(request)
+        if response.ok:
+            demisto.debug('authenticated successfully')
+            self.access_token = response.json()['access_token']
+            self.request.headers['Authorization'] = f'Bearer {self.access_token}'
+        else:
+            demisto.debug(f'authentication failed: {response.json()}')
+
+
+class CyberArkIdentityGetEvents(IntegrationGetEvents):
+    client: CyberArkIdentityEventsClient
+
+    @staticmethod
+    def get_last_run_ids(events: list) -> list:
+        return [event.get('ID') for event in events]
+
+    @staticmethod
+    def get_last_run_time(events: list) -> str:
+        # The date is in timestamp format and looks like {'WhenOccurred': '/Date(1651483379362)/'}
+        last_timestamp = max([int(e.get('WhenOccurred', '').removesuffix(')/').removeprefix('/Date(')) for e in events])
+
+        return datetime.utcfromtimestamp(last_timestamp / 1000).strftime(DATETIME_FORMAT)
+
+    def get_last_run(self, events: list) -> dict:  # type: ignore
+        return {'from': self.get_last_run_time(events), 'ids': self.get_last_run_ids(events)}
+
+    def _iter_events(self):
+        self.client.authenticate()
+
+        result = self.client.call(self.client.request).json()['Result']
+
+        if events := result.get('Results'):
+            fetched_events_ids = demisto.getLastRun().get('ids', [])
+            yield [event.get('Row') for event in events if event.get('Row', {}).get('ID') not in fetched_events_ids]
+
+
+def get_request_params(**kwargs: dict) -> dict:
+    fetch_from = str(kwargs.get('from', '3 days'))
+    default_from_day = datetime.now() - timedelta(days=3)
+    from_time = datetime.strftime(dateparser.parse(fetch_from, settings={'TIMEZONE': 'UTC'}) or default_from_day, DATETIME_FORMAT)
+
+    params = {
+        'url': f'{str(kwargs.get("url", "")).removesuffix("/")}/RedRock/Query',
+        'data': json.dumps({
+            "Script": f"Select {', '.join(EVENT_FIELDS)} from Event where WhenOccurred > '{from_time}'"
+        }),
+        'verify': not kwargs.get('insecure')
+    }
+    return params
+
+
+def main(command: str, demisto_params: dict):
+    credentials = Credentials(**demisto_params.get('credentials', {}))
+    options = CyberArkIdentityEventsOptions(**demisto_params)
+    request_params = get_request_params(**demisto_params)
+    request = CyberArkIdentityEventsRequest(**request_params)
+    client = CyberArkIdentityEventsClient(request, options, credentials)
+    get_events = CyberArkIdentityGetEvents(client, options)
+
+    try:
+        if command == 'test-module':
+            get_events.run()
+            demisto.results('ok')
+
+        if command in ('fetch-events', 'cyberarkidentity-get-events'):
+            events = get_events.run()
+
+            if events:
+                last_run = get_events.get_last_run(events)
+                demisto.debug(f'Set last run to {last_run}')
+                demisto.setLastRun(last_run)
```

Comment:
we don't want to set the last run only if the command is  `cyberarkidentity-get-events` there might be a possibility that `should_push_events` is false and then we will not save some events